### PR TITLE
Concf 801

### PR DESCRIPTION
**Pull request by justnpT**
Acceptance Criteria:
* check if tap on elements from different namespaces redirects to correct pages
** check if article loads inside mercury
** check if other items redirect outside mercury (except category) - check class on body, skin JS variable or more ways to check it
* test are executed daily

Algorithm:
* open http://mercurycc.wikia.com/main/Category:Articles
* check icon on image is #namespace-article
* tap on first element
* check url pattern /wiki/

* open http://mercurycc.wikia.com/main/category/Blog_posts
* check icon on image is #namespace-blog
* tap on first element
* check url pattern /wiki/User_blog:

* open http://mercurycc.wikia.com/main/category/Images
* check icon on image is #namespace-image
* tap on first element
* check url pattern /wiki/File:

* open http://mercurycc.wikia.com/main/category/Videos
* check icon on image is #namespace-video
* tap on first element
* check url pattern /wiki/File:

### CODE REVIEW

**Comment by tomasznapieralski** [comment]
LGTM

File: src/test/java/com/wikia/webdriver/common/contentpatterns/MercurySubpages.java [hunk 1]
```diff
@@ -29,6 +29,11 @@ private MercurySubpages() {}
 
   // Articles prepared for mercurycc.wikia.com wiki (with FC, CC, TA and TV)
   public static final String CC_MAIN_PAGE = "Mercury_CC_Wikia";
+  // Categories in mercurycc.wikia.com
+  public static final String CC_CATEGORY_ARTICLES = "main/category/Articles";
+  public static final String CC_CATEGORY_BLOGS = "main/category/Blog_posts";
+  public static final String CC_CATEGORY_IMAGES = "main/category/Images";
+  public static final String CC_CATEGORY_VIDEOS = "main/category/Videos";
```
Comment (bognix) [inline src/test/java/com/wikia/webdriver/common/contentpatterns/MercurySubpages.java]: how about creating Class for all constants related to curated main pages? so it is nicely grouped.

File: src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java [hunk 2]
```diff
@@ -82,6 +82,13 @@ public CuratedMainPagePageObject openCuratedMainPage(String wikiURL, String main
     return new CuratedMainPagePageObject(driver);
   }
 
+  public CuratedContentPageObject openCuratedContentPage(String wikiURL, String path) {
+    getUrl(wikiURL + path + "?cb=" + DateTime.now().getMillis());
```
Comment (bognix) [inline src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/BasePageObject.java]: you can use url builder here.

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java [hunk 3]
```diff
@@ -16,16 +16,19 @@
  */
 public class NavigationTests extends NewTestTemplate {
 
+  private static final String ROOT_PATH = "/";
+  private static final String ROOT_PATH_SECTION = "/main/section/";
+  private static final String ROOT_PATH_CATEGORY = "/main/category/";
+  private static final String ROOT_ARTICLE_PATH = "/wiki/";
```
Comment (bognix) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java]: isn't this defined somewhere already?

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java [hunk 4]
```diff
@@ -16,16 +16,19 @@
  */
 public class NavigationTests extends NewTestTemplate {
 
+  private static final String ROOT_PATH = "/";
+  private static final String ROOT_PATH_SECTION = "/main/section/";
+  private static final String ROOT_PATH_CATEGORY = "/main/category/";
```
Comment (bognix) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java]: shouldn't it be defined inside of class MercurySubpages (or similar class related with curated main pages)

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java [hunk 5]
```diff
@@ -76,4 +79,38 @@ public void MercuryCuratedNavigationTests_002_navigateThroughSection() {
...
+        .isImageIconVisible()
+        .clickOnCuratedContentElementByIndex(0)
+        .waitForLoadingSpinnerToFinishReloadingPage();
+    UrlChecker.isPathContainedInCurrentUrl(driver, ROOT_ARTICLE_PATH);
+
+    category
+        .openCuratedContentPage(wikiURL, MercurySubpages.CC_CATEGORY_BLOGS)
+        .isVideoIconVisible()
+        .clickOnCuratedContentElementByIndex(0)
+        .waitForLoadingSpinnerToFinishReloadingPage();
```
Comment (bognix) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java]: this page should be called "waitForPageToLoad()" - base on this name you can get an impression that spinner is doing the reloading.

File: src/test/java/com/wikia/webdriver/common/contentpatterns/MercurySubpages.java (hunk 1 above)
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/common/contentpatterns/MercurySubpages.java]: We have two files `MercuryWikis` for all wikis we are using in tests
and `MercurySubpages` for all articles we are using.
It is grouped enough

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java (hunk 3 above)
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java]: nope

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java (hunk 4 above)
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java]: It is only pattern for test condition it should be there

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java (hunk 5 above)
Comment (tomasznapieralski) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java]: or like `waitForAsyncPageLoad()`

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java [hunk 6]
```diff
@@ -76,4 +79,38 @@ public void MercuryCuratedNavigationTests_002_navigateThroughSection() {
...
+    UrlChecker.isPathContainedInCurrentUrl(driver, ROOT_ARTICLE_PATH);
+
+    category
+        .openCuratedContentPage(wikiURL, MercurySubpages.CC_CATEGORY_BLOGS)
+        .isImageIconVisible()
+        .clickOnCuratedContentElementByIndex(0)
+        .waitForLoadingSpinnerToFinishReloadingPage();
+    UrlChecker.isPathContainedInCurrentUrl(driver, ROOT_ARTICLE_PATH);
```
Comment (bognix) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java]: you want to check if you are on article? How about checking if ".article-wrapper" selector is on page instead? /wiki/ can be also edit page, not only article

File: src/test/java/com/wikia/webdriver/common/contentpatterns/MercurySubpages.java (hunk 1 above)
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/common/contentpatterns/MercurySubpages.java]: It's classification topic again 
Current solution uses wikis as core classifier for the content 
Bogna proposition is to have functionality as the core classifier 

Intuitively Bogna proposition is better in terms of being independent from tested environment
On the other hand the tested data environment is part of test composition. That's the direction we want to follow

I'd leave it for open discussion

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java (hunk 4 above)
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java]: I agree with Bogna that these paths could be reused in other tests
Test is not the best place for storing validated arguments
I'll prepare class for static curated content paths

File: src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java (hunk 6 above)
Comment (justnpT) [inline src/test/java/com/wikia/webdriver/testcases/mercurytests/curatedcontenttests/NavigationTests.java]: I'll extend the test with the selector check

**Comment by bognix** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
LGTM

**Comment by bognix** [comment]
LGTM

**Pull request merged**